cpp/src/parquet/reader_test.cc

```diff
@@ -502,6 +514,160 @@ TEST_F(TestLocalFile, OpenWithMetadata) {
   ASSERT_EQ(metadata.get(), reader2->metadata().get());
 }
 
+// https://github.com/google/googletest/pull/2904 not available in our version of
+// gtest/gmock
```

Comment:
The user might have installed a more recent version of GTest, so I would guard this with a `#ifdef`.

Reply:
The code is move-ahead from L861-L870 in the same file. Origin tests uses it.

Though I think I'd like to move it to parquet test tools